### ISSUE TITLE
Need a gson-2.10.1 jar build without module-info.class

### ISSUE THREAD

**Issue by shubham-takale-sp** [google/gson#2488]
# Gson version
 2.10.1

# Java / Android version
 SAP JVM8

# Used tools

- [ ] Maven; version: 
- [x] Gradle; version: 
- [ ] ProGuard (attach the configuration file please); version: 
- [ ] ...

# Description

We are having a custom java application which is using gson-2.10.1 as a dependency. This application is deployed on SAP Server which uses SAP's JVM8. During deployment, the server is not able to parse the module-info.class available in gson jar. **So we need a gson-2.10.1.jar in non-JPMS format(without the module-info.class file).**

## Expected behavior

Our custom application gets successfully deployed on SAP Server using gson-2.10.1.jar.

## Actual behavior

Custom application deployment is failing.

# Reproduction steps

NA

# Exception stack trace

```
Caused by: com.sap.engine.library.bytecode.cf.CFException: Invalid constant pool tag, 19 at index 4.
	at com.sap.engine.library.bytecode.cf.CFParser.parse0(CFParser.java:386)
	at com.sap.engine.library.bytecode.cf.CFParser.parse(CFParser.java:307)
	at com.sap.lib.javalang.tool.impl.ClassInfoReaderImpl.parseClassInformation(ClassInfoReaderImpl.java:336)
	at com.sap.lib.javalang.tool.impl.ClassInfoReaderImpl.traverseZipFile(ClassInfoReaderImpl.java:287)
	at com.sap.lib.javalang.tool.impl.ClassInfoReaderImpl.readInternal(ClassInfoReaderImpl.java:105)
	at com.sap.lib.javalang.tool.impl.ClassInfoReaderImpl.traverseFolder(ClassInfoReaderImpl.java:175)
	at com.sap.lib.javalang.tool.impl.ClassInfoReaderImpl.readInternal(ClassInfoReaderImpl.java:103)
	at com.sap.lib.javalang.tool.impl.ClassInfoReaderImpl.traverseFolder(ClassInfoReaderImpl.java:175)
	at com.sap.lib.javalang.tool.impl.ClassInfoReaderImpl.readInternal(ClassInfoReaderImpl.java:103)
	at com.sap.lib.javalang.tool.impl.ClassInfoReaderImpl.traverseWarFile(ClassInfoReaderImpl.java:196)
	at com.sap.lib.javalang.tool.impl.ClassInfoReaderImpl.traverseFolder(ClassInfoReaderImpl.java:165)
	at com.sap.lib.javalang.tool.impl.ClassInfoReaderImpl.readInternal(ClassInfoReaderImpl.java:103)
	at com.sap.lib.javalang.tool.impl.ClassInfoReaderImpl.traverseFolder(ClassInfoReaderImpl.java:173)
	at com.sap.lib.javalang.tool.impl.ClassInfoReaderImpl.readInternal(ClassInfoReaderImpl.java:103)
	at com.sap.lib.javalang.tool.impl.ClassInfoReaderImpl.read(ClassInfoReaderImpl.java:95)
	at com.sap.engine.services.deploy.ear.jar.DplArchiveReader.parseAnnotations(DplArchiveReader.java:62)
	at com.sap.engine.services.deploy.ear.jar.EARReader.read(EARReader.java:243)
	at com.sap.engine.services.deploy.server.application.DeployUtilTransaction.prepareNewDeploymentInfo(DeployUtilTransaction.java:251)
	at com.sap.engine.services.deploy.server.application.DeploymentTransaction.begin(DeploymentTransaction.java:199)
	at com.sap.engine.services.deploy.server.application.ApplicationTransaction.makeAllPhasesOnOneServer(ApplicationTransaction.java:423)
	... 16 more
```

**Comment by Marcono1234** [comment]
Gson intentionally already places the `module-info.class` file in the JAR under `META-INF/versions/9` to reduce the risk of causing issues for Java 8 and Android users. This approach seems to be somewhat common and is for example also used by [Jackson](https://github.com/FasterXML/jackson-bom/blob/a57627ddc17ef06c9d7b4de673aef8f51a69ed5c/base/pom.xml#L157-L185) and [Log4j 2](https://issues.apache.org/jira/browse/LOG4J2-2271).

By now a lot of tools with Java 8 support have been adjusted to ignore the `module-info.class` file. Could you please check if there is a newer version of `com.sap.engine.library.bytecode.cf.CFParser` which ignores the `module-info.class` file?

**Comment by shubham-takale-sp** [comment]
I checked with SAP team, they do not have such release available as of now. So, in this situation would you help us in getting the gson-2.10.1 jar not in JPMS format i.e., without module-info.class file.

**Comment by eamonnmcmanus** [comment]
Do you have the option of using an older version of Gson? Any version earlier than 2.8.6 would work.

Alternatively you might be able to make a modified 2.10.1 jar yourself, by starting with the [released `gson-2.10.1.jar`](https://repo1.maven.org/maven2/com/google/code/gson/gson/2.10.1/gson-2.10.1.jar) and running this:
```
zip -d gson-2.10.1.jar META-INF/versions/9/module-info.class
```

I think it's unlikely that we're going to make a separate release artifact for your situation. 2.8.6 was released four years ago with `module-info.class` and we haven't needed that separate artifact so far.

**Comment by Marcono1234** [comment]
Have changed the label from "bug" to "enhancement" since the `module-info.class` file is intentionally included in the JAR, so this is not really a bug but rather a feature request.
However, what Éamonn wrote above still applies I think; it is unlikely that a separate JAR without `module-info.class` file will be published.
